ipn/ipnlocal/local.go

```diff
@@ -7423,6 +7424,9 @@ func suggestExitNodeUsingDERP(report *netcheck.Report, nb *nodeBackend, prevSugg
 		}
 	}
 	bestCandidates := pickWeighted(pickFrom)
+	if len(bestCandidates) == 0 {
+		return res, ErrNoSuitableCandidates
+	}
```

Comment:
❓ Question (blocking): Returning an error here doesn’t seem to match what we do when `candidates` is an empty array or only has one value. For the former, we don’t complain at all. For the latter, we always return that result. So if there are no `pickFrom` options, because there are no distances, shouldn’t we just pick any one of the candidates? So could we set `bestCandidates = candidates`?